web3/utils/module_testing/eth_module.py

```diff
@@ -213,6 +213,145 @@ def test_eth_sendTransaction(self, web3, unlocked_account):
         assert txn['gas'] == 21000
         assert txn['gasPrice'] == txn_params['gas_price']
 
+    def test_eth_replaceTransaction(self, web3, unlocked_account):
+        txn_params = {
+            'from': unlocked_account,
+            'to': unlocked_account,
+            'value': 1,
+            'gas': 21000,
+            'gasPrice': web3.eth.gasPrice,
+        }
+        txn_hash = web3.eth.sendTransaction(txn_params)
+
+        txn_params['gasPrice'] = web3.eth.gasPrice * 2
+        replace_txn_hash = web3.eth.replaceTransaction(txn_hash, txn_params)
+        replace_txn = web3.eth.getTransaction(replace_txn_hash)
```

Comment:
I am pretty sure that this should work, too, after replacement:

```py
assert web3.eth.getTransaction(txn_hash) is None
```

If it works in other nodes, but fails in eth-tester, then please make a tracking issue in eth-tester.

Reply:
On initial thought, I think the transaction would exist until the block started mining regardless of node in which case it could make the tests really strange to add this.  I'll check the behaviour.

Reply:
Tried with geth, indeed the old transaction is removed 🙂 
Raised https://github.com/ethereum/eth-tester/issues/57 
Can have a look at it in the weekend